components/brave_new_tab_ui/reducers/cryptoDotCom_reducer.ts

```diff
@@ -15,51 +15,54 @@ function performSideEffect (fn: () => void): void {
   window.setTimeout(() => fn(), 0)
 }
 
+function reducePairs (rawPairs: SupportedPair[]) {
+  return rawPairs.reduce((pairs: object, currPair: SupportedPair) => {
+    const { base, pair } = currPair
+    pairs[base] = pairs[base]
+      ? [...pairs[base], pair]
+      : [pair]
+    return pairs
+  }, {})
+}
+
 const cryptoDotComReducer: Reducer<NewTab.State | undefined> = (state: NewTab.State, action) => {
   const payload = action.payload
 
   switch (action.type) {
-    case types.ON_TOTAL_PRICE_OPT_IN:
-      state = { ...state }
-      state.cryptoDotComState.optInTotal = true
-      break
-
     case types.ON_BTC_PRICE_OPT_IN:
       state = { ...state }
       state.cryptoDotComState.optInBTCPrice = true
       break
 
-    case types.SET_TICKER_PRICES:
+    case types.SET_MARKETS_REQUESTED:
```

Comment:
As mentioned `MARKETS_REQUESTED` (`SET_` is rarely a good prefix for an action) sounds like an event which should fire off API calls to get data. And something like `MARKET_DATA_UPDATED` would be appropriate here.

Reply:
changed